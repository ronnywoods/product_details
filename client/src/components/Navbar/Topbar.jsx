import React, { Component } from 'react';
import DropdownMenu from './DropdownMenu.jsx';
import menu from './data.js';

class Topbar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shop: false,
    }
  }

  render() {
    let cart = this.props.cart === 0 ? '' : this.props.cart;
    let iconClass = this.props.cart === 0 ? 'material-icons md-light' : 'material-icons md-light cartIcon';
    let cartClass = this.props.cart === 0 ? '' : 'cart';
    return (
      <div className="sticky-top">
        <div className="topBar">
          <div className="help white">
            <li> HELP</li>
            <li> | </li>
            <li> ORDER TRACKER AND RETURNS</li>
            <li> | </li>
            <li id="fbold"> NEWSLETTER SIGNUP</li>
            <li> LOGIN</li>
            <li></li>
            <i className="material-icons md-light">person</i>
          </div>
        </div>
        <div className="bBar">
          <div className="row width-100">
            <img src="https://s3-us-west-1.amazonaws.com/adidas-shoe/adidas-logo-white.png" width="75" height="75" className="nav-logo d-inline-block align-top"></img>
            <div className="choices">
              <div className="dropdown">
                <li className="choice-menu white" id="men-dropdown" data-toggle="dropdown"> MENS</li>
                <div className="dropdown-menu bar-menu" aria-labelledby="dropdown-men">
                  <div className="menu-men">
                    <DropdownMenu title={'FEATURED'} options={menu.men.featured} />
                    <DropdownMenu title={'SHOES'} options={menu.men.shoes} />
                    <DropdownMenu title={'CLOTHING'} options={menu.men.clothing} />
                    <DropdownMenu title={'ACCESORIES'} options={menu.men.accessories} />
                    <DropdownMenu title={'SPORTS'} options={menu.men.sports} />
                  </div>
                </div>
              </div>
              <div className="dropdown">
                <li className="choice-menu white" id="women-dropdown" data-toggle="dropdown"> WOMENS</li>
                <div className="dropdown-menu bar-menu" aria-labelledby="dropdown-women">
                  <div className="menu-men">
                    <DropdownMenu title={'FEATURED'} options={menu.women.featured} />
                    <DropdownMenu title={'SHOES'} options={menu.women.shoes} />
                    <DropdownMenu title={'CLOTHING'} options={menu.women.clothing} />
                    <DropdownMenu title={'ACCESORIES'} options={menu.women.accessories} />
                    <DropdownMenu title={'SPORTS'} options={menu.women.sports} />
                  </div>
                </div>
              </div>
              <div className="dropdown">
                <li className="choice-menu white" id="kids-dropdown" data-toggle="dropdown"> KIDS</li>
                <div className="dropdown-menu bar-menu" aria-labelledby="dropdown-kids">
                  <div className="menu-men">
                    <DropdownMenu title={'FEATURED'} options={menu.kids.featured} />
                    <DropdownMenu title={'YOUTH\n(8-14)'} options={menu.kids.youth} />
                    <DropdownMenu title={'CHILDREN\n(4-8)'} options={menu.kids.children} />
                    <DropdownMenu title={'BABY\n(0-4)'} options={menu.kids.baby} />
                    <DropdownMenu title={'ACCESSORIES'} options={menu.kids.accessories} />
                  </div>
                </div>
              </div>
              <li className="gray"> | </li>
              <li className="choice-menu white"> SPORTS</li>
              <li className="choice-menu white"> BRANDS</li>
            </div>
          </div>
          <div className="search white">
            <span id="search-box"><i className="material-icons">search</i></span>
            <input type="text" placeholder="search"></input>
            <li></li>
            <div className={cartClass}><i className={iconClass}>shopping_cart</i><span className="cartNum">{cart}</span></div>
          </div>
        </div>
        <div className="shipping-bar">
          <h6>
            <i className="material-icons delivery" height="24" width="24">local_shipping</i>
            <span className="underline">FREE SHIPPING AND RETURNS</span>
            <img className="creators"src="https://s3-us-west-1.amazonaws.com/adidas-shoe/adidas-icon-creator.png" width="24" height="24"></img>
            <span className="underline">CREATORS GET REWARDED - JOIN THE NEW ADIDAS CREATOR CLUB</span>
          </h6>
        </div>
      </div>
    )
  }
}

export default Topbar;
