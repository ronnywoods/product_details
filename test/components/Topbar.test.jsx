import React from 'react';
import { shallow } from 'enzyme';

import Topbar from '../../client/src/components/Navbar/Topbar';

describe('Topbar/Navbar Component', () => {
  const component = shallow(<Topbar cart={0} />);
  const component2 = shallow(<Topbar cart={2} />);

  describe('When cart is empty', () => {
    it('should match snapshot when cart is empty', () => {
      expect(component).toMatchSnapshot();
    });

    it('should not contain cart numbers', () => {
      expect(component.find('.cartNum').text()).toBe('');
    });
  });

  describe('When cart has contents', () => {
    it('should match snapshot when cart has content', () => {
      expect(component2).toMatchSnapshot();
    });

    it('should display cart number properly', () => {
      expect(component2.find('.cart')).toExist();
      expect(component2.find('.cartNum').text()).toBe('2');
    });
  });

});
