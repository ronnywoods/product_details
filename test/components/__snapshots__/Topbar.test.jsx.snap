// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar/Navbar Component When cart has contents should match snapshot when cart has content 1`] = `
<div
  className="sticky-top"
>
  <div
    className="topBar"
  >
    <div
      className="help white"
    >
      <li>
         HELP
      </li>
      <li>
         | 
      </li>
      <li>
         ORDER TRACKER AND RETURNS
      </li>
      <li>
         | 
      </li>
      <li
        id="fbold"
      >
         NEWSLETTER SIGNUP
      </li>
      <li>
         LOGIN
      </li>
      <li />
      <i
        className="material-icons md-light"
      >
        person
      </i>
    </div>
  </div>
  <div
    className="bBar"
  >
    <div
      className="row width-100"
    >
      <img
        className="nav-logo d-inline-block align-top"
        height="75"
        src="https://s3-us-west-1.amazonaws.com/adidas-shoe/adidas-logo-white.png"
        width="75"
      />
      <div
        className="choices"
      >
        <div
          className="dropdown"
        >
          <li
            className="choice-menu white"
            data-toggle="dropdown"
            id="men-dropdown"
          >
             MENS
          </li>
          <div
            aria-labelledby="dropdown-men"
            className="dropdown-menu bar-menu"
          >
            <div
              className="menu-men"
            >
              <DropdownMenu
                options={
                  Array [
                    "New Arrivals",
                    "Best Sellers",
                    "Release Dates",
                    "____",
                    "YEEZY",
                    "Ultraboost",
                    "Harden Vol.3",
                    "NMD",
                    "Statement Sneakers",
                    "Workout Essentials",
                    "____",
                    "Sale",
                  ]
                }
                title="FEATURED"
              />
              <DropdownMenu
                options={
                  Array [
                    "Originals",
                    "Running",
                    "Soccer",
                    "Basketball",
                    "Football",
                    "Skateboarding",
                    "Workout",
                    "Essentials",
                    "Sandals & Sneakers",
                    "Hiking",
                    "Golf",
                    "Tennis",
                    "Basketball",
                    "Sale Shoes",
                  ]
                }
                title="SHOES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Pants",
                    "Hoodies & Sweatshirts",
                    "Jackets & Vests",
                    "Track Suits",
                    "Short Sleeve Shirts",
                    "Graphic T-Shirts",
                    "Long Sleeve Shirts",
                    "Jerseys",
                    "Tights",
                    "Shorts",
                    "Tank Tops",
                    "Sale Apparel",
                  ]
                }
                title="CLOTHING"
              />
              <DropdownMenu
                options={
                  Array [
                    "Bags & Backpacks",
                    "Hats & Beanies ",
                    "Socks",
                    "Phone Cases",
                    "Sunglasses",
                    "Balls",
                    "Watches",
                    "Gloves",
                    "All Accessories",
                    "Sale Accessories",
                  ]
                }
                title="ACCESORIES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Running",
                    "Soccer",
                    "Basketball",
                    "Football",
                    "Skateboarding",
                    "Training",
                    "Outdoor",
                    "Tennis",
                    "Golf",
                    "Hockey",
                    "Baseball",
                    "Lacrosse",
                    "Volleyball",
                    "Weightlifting",
                    "College Gear",
                  ]
                }
                title="SPORTS"
              />
            </div>
          </div>
        </div>
        <div
          className="dropdown"
        >
          <li
            className="choice-menu white"
            data-toggle="dropdown"
            id="women-dropdown"
          >
             WOMENS
          </li>
          <div
            aria-labelledby="dropdown-women"
            className="dropdown-menu bar-menu"
          >
            <div
              className="menu-men"
            >
              <DropdownMenu
                options={
                  Array [
                    "New Arrivals",
                    "Best Sellers",
                    "Release Dates",
                    "____",
                    "YEEZY",
                    "Ultraboost",
                    "NMD",
                    "Statement Sneakers",
                    "Workout Essentials",
                    "____",
                    "Sale",
                  ]
                }
                title="FEATURED"
              />
              <DropdownMenu
                options={
                  Array [
                    "Originals",
                    "Running",
                    "Workout",
                    "Essentials",
                    "Sandals & Sneakers",
                    "Hiking",
                    "Golf",
                    "Soccer",
                    "Tennis",
                    "Volleyball",
                    "Sale Shoes",
                  ]
                }
                title="SHOES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Hoodies & Sweatshirts",
                    "Pants",
                    "Leggings & Tights",
                    "Sports Bras",
                    "Jackets & Vests",
                    "Track Suits",
                    "Long Sleeve Shirts",
                    "Dresses & Skirts",
                    "Shorts",
                    "Tank Tops",
                    "Sale Apparel",
                  ]
                }
                title="CLOTHING"
              />
              <DropdownMenu
                options={
                  Array [
                    "Bags & Backpacks",
                    "Hats & Beanies ",
                    "Socks",
                    "Phone Cases",
                    "Sunglasses",
                    "Balls",
                    "Watches",
                    "Gloves",
                    "All Accessories",
                    "Sale Accessories",
                  ]
                }
                title="ACCESORIES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Running",
                    "Training",
                    "Soccer",
                    "Basketball",
                    "Skateboarding",
                    "Outdoor",
                    "Tennis",
                    "Golf",
                    "Yoga",
                    "Lacrosse",
                    "Volleyball",
                    "Hockey",
                    "College Gear",
                  ]
                }
                title="SPORTS"
              />
            </div>
          </div>
        </div>
        <div
          className="dropdown"
        >
          <li
            className="choice-menu white"
            data-toggle="dropdown"
            id="kids-dropdown"
          >
             KIDS
          </li>
          <div
            aria-labelledby="dropdown-kids"
            className="dropdown-menu bar-menu"
          >
            <div
              className="menu-men"
            >
              <DropdownMenu
                options={
                  Array [
                    "New Arrivals",
                    "Best Sellers",
                    "____",
                    "Ultraboost",
                    "NMD",
                    "adicolor",
                    "Statement Sneakers",
                    "____",
                    "Sale",
                  ]
                }
                title="FEATURED"
              />
              <DropdownMenu
                options={
                  Array [
                    "Boys Shoes",
                    "Girls Shoes",
                    "Original Shoes",
                    "Sport Shoes",
                    "____",
                    "Boys Clothing",
                    "Girls Clothing",
                    "Hoodies & SweatShirts",
                    "Pants & Shorts",
                    "____",
                    "Sale",
                  ]
                }
                title="YOUTH
(8-14)"
              />
              <DropdownMenu
                options={
                  Array [
                    "Boys Shoes",
                    "Girls Shoes",
                    "Original Shoes",
                    "Sport Shoes",
                    "____",
                    "Boys Clothing",
                    "Girls Clothing",
                    "____",
                    "Sale",
                  ]
                }
                title="CHILDREN
(4-8)"
              />
              <DropdownMenu
                options={
                  Array [
                    "Shoes",
                    "Clothing",
                    "____",
                    "Sale",
                  ]
                }
                title="BABY
(0-4)"
              />
              <DropdownMenu
                options={
                  Array [
                    "Socks",
                    "Hats & Beanies",
                    "Balls",
                  ]
                }
                title="ACCESSORIES"
              />
            </div>
          </div>
        </div>
        <li
          className="gray"
        >
           | 
        </li>
        <li
          className="choice-menu white"
        >
           SPORTS
        </li>
        <li
          className="choice-menu white"
        >
           BRANDS
        </li>
      </div>
    </div>
    <div
      className="search white"
    >
      <span
        id="search-box"
      >
        <i
          className="material-icons"
        >
          search
        </i>
      </span>
      <input
        placeholder="search"
        type="text"
      />
      <li />
      <div
        className="cart"
      >
        <i
          className="material-icons md-light cartIcon"
        >
          shopping_cart
        </i>
        <span
          className="cartNum"
        >
          2
        </span>
      </div>
    </div>
  </div>
  <div
    className="shipping-bar"
  >
    <h6>
      <i
        className="material-icons delivery"
        height="24"
        width="24"
      >
        local_shipping
      </i>
      <span
        className="underline"
      >
        FREE SHIPPING AND RETURNS
      </span>
      <img
        className="creators"
        height="24"
        src="https://s3-us-west-1.amazonaws.com/adidas-shoe/adidas-icon-creator.png"
        width="24"
      />
      <span
        className="underline"
      >
        CREATORS GET REWARDED - JOIN THE NEW ADIDAS CREATOR CLUB
      </span>
    </h6>
  </div>
</div>
`;

exports[`Topbar/Navbar Component When cart is empty should match snapshot when cart is empty 1`] = `
<div
  className="sticky-top"
>
  <div
    className="topBar"
  >
    <div
      className="help white"
    >
      <li>
         HELP
      </li>
      <li>
         | 
      </li>
      <li>
         ORDER TRACKER AND RETURNS
      </li>
      <li>
         | 
      </li>
      <li
        id="fbold"
      >
         NEWSLETTER SIGNUP
      </li>
      <li>
         LOGIN
      </li>
      <li />
      <i
        className="material-icons md-light"
      >
        person
      </i>
    </div>
  </div>
  <div
    className="bBar"
  >
    <div
      className="row width-100"
    >
      <img
        className="nav-logo d-inline-block align-top"
        height="75"
        src="https://s3-us-west-1.amazonaws.com/adidas-shoe/adidas-logo-white.png"
        width="75"
      />
      <div
        className="choices"
      >
        <div
          className="dropdown"
        >
          <li
            className="choice-menu white"
            data-toggle="dropdown"
            id="men-dropdown"
          >
             MENS
          </li>
          <div
            aria-labelledby="dropdown-men"
            className="dropdown-menu bar-menu"
          >
            <div
              className="menu-men"
            >
              <DropdownMenu
                options={
                  Array [
                    "New Arrivals",
                    "Best Sellers",
                    "Release Dates",
                    "____",
                    "YEEZY",
                    "Ultraboost",
                    "Harden Vol.3",
                    "NMD",
                    "Statement Sneakers",
                    "Workout Essentials",
                    "____",
                    "Sale",
                  ]
                }
                title="FEATURED"
              />
              <DropdownMenu
                options={
                  Array [
                    "Originals",
                    "Running",
                    "Soccer",
                    "Basketball",
                    "Football",
                    "Skateboarding",
                    "Workout",
                    "Essentials",
                    "Sandals & Sneakers",
                    "Hiking",
                    "Golf",
                    "Tennis",
                    "Basketball",
                    "Sale Shoes",
                  ]
                }
                title="SHOES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Pants",
                    "Hoodies & Sweatshirts",
                    "Jackets & Vests",
                    "Track Suits",
                    "Short Sleeve Shirts",
                    "Graphic T-Shirts",
                    "Long Sleeve Shirts",
                    "Jerseys",
                    "Tights",
                    "Shorts",
                    "Tank Tops",
                    "Sale Apparel",
                  ]
                }
                title="CLOTHING"
              />
              <DropdownMenu
                options={
                  Array [
                    "Bags & Backpacks",
                    "Hats & Beanies ",
                    "Socks",
                    "Phone Cases",
                    "Sunglasses",
                    "Balls",
                    "Watches",
                    "Gloves",
                    "All Accessories",
                    "Sale Accessories",
                  ]
                }
                title="ACCESORIES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Running",
                    "Soccer",
                    "Basketball",
                    "Football",
                    "Skateboarding",
                    "Training",
                    "Outdoor",
                    "Tennis",
                    "Golf",
                    "Hockey",
                    "Baseball",
                    "Lacrosse",
                    "Volleyball",
                    "Weightlifting",
                    "College Gear",
                  ]
                }
                title="SPORTS"
              />
            </div>
          </div>
        </div>
        <div
          className="dropdown"
        >
          <li
            className="choice-menu white"
            data-toggle="dropdown"
            id="women-dropdown"
          >
             WOMENS
          </li>
          <div
            aria-labelledby="dropdown-women"
            className="dropdown-menu bar-menu"
          >
            <div
              className="menu-men"
            >
              <DropdownMenu
                options={
                  Array [
                    "New Arrivals",
                    "Best Sellers",
                    "Release Dates",
                    "____",
                    "YEEZY",
                    "Ultraboost",
                    "NMD",
                    "Statement Sneakers",
                    "Workout Essentials",
                    "____",
                    "Sale",
                  ]
                }
                title="FEATURED"
              />
              <DropdownMenu
                options={
                  Array [
                    "Originals",
                    "Running",
                    "Workout",
                    "Essentials",
                    "Sandals & Sneakers",
                    "Hiking",
                    "Golf",
                    "Soccer",
                    "Tennis",
                    "Volleyball",
                    "Sale Shoes",
                  ]
                }
                title="SHOES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Hoodies & Sweatshirts",
                    "Pants",
                    "Leggings & Tights",
                    "Sports Bras",
                    "Jackets & Vests",
                    "Track Suits",
                    "Long Sleeve Shirts",
                    "Dresses & Skirts",
                    "Shorts",
                    "Tank Tops",
                    "Sale Apparel",
                  ]
                }
                title="CLOTHING"
              />
              <DropdownMenu
                options={
                  Array [
                    "Bags & Backpacks",
                    "Hats & Beanies ",
                    "Socks",
                    "Phone Cases",
                    "Sunglasses",
                    "Balls",
                    "Watches",
                    "Gloves",
                    "All Accessories",
                    "Sale Accessories",
                  ]
                }
                title="ACCESORIES"
              />
              <DropdownMenu
                options={
                  Array [
                    "Running",
                    "Training",
                    "Soccer",
                    "Basketball",
                    "Skateboarding",
                    "Outdoor",
                    "Tennis",
                    "Golf",
                    "Yoga",
                    "Lacrosse",
                    "Volleyball",
                    "Hockey",
                    "College Gear",
                  ]
                }
                title="SPORTS"
              />
            </div>
          </div>
        </div>
        <div
          className="dropdown"
        >
          <li
            className="choice-menu white"
            data-toggle="dropdown"
            id="kids-dropdown"
          >
             KIDS
          </li>
          <div
            aria-labelledby="dropdown-kids"
            className="dropdown-menu bar-menu"
          >
            <div
              className="menu-men"
            >
              <DropdownMenu
                options={
                  Array [
                    "New Arrivals",
                    "Best Sellers",
                    "____",
                    "Ultraboost",
                    "NMD",
                    "adicolor",
                    "Statement Sneakers",
                    "____",
                    "Sale",
                  ]
                }
                title="FEATURED"
              />
              <DropdownMenu
                options={
                  Array [
                    "Boys Shoes",
                    "Girls Shoes",
                    "Original Shoes",
                    "Sport Shoes",
                    "____",
                    "Boys Clothing",
                    "Girls Clothing",
                    "Hoodies & SweatShirts",
                    "Pants & Shorts",
                    "____",
                    "Sale",
                  ]
                }
                title="YOUTH
(8-14)"
              />
              <DropdownMenu
                options={
                  Array [
                    "Boys Shoes",
                    "Girls Shoes",
                    "Original Shoes",
                    "Sport Shoes",
                    "____",
                    "Boys Clothing",
                    "Girls Clothing",
                    "____",
                    "Sale",
                  ]
                }
                title="CHILDREN
(4-8)"
              />
              <DropdownMenu
                options={
                  Array [
                    "Shoes",
                    "Clothing",
                    "____",
                    "Sale",
                  ]
                }
                title="BABY
(0-4)"
              />
              <DropdownMenu
                options={
                  Array [
                    "Socks",
                    "Hats & Beanies",
                    "Balls",
                  ]
                }
                title="ACCESSORIES"
              />
            </div>
          </div>
        </div>
        <li
          className="gray"
        >
           | 
        </li>
        <li
          className="choice-menu white"
        >
           SPORTS
        </li>
        <li
          className="choice-menu white"
        >
           BRANDS
        </li>
      </div>
    </div>
    <div
      className="search white"
    >
      <span
        id="search-box"
      >
        <i
          className="material-icons"
        >
          search
        </i>
      </span>
      <input
        placeholder="search"
        type="text"
      />
      <li />
      <div
        className=""
      >
        <i
          className="material-icons md-light"
        >
          shopping_cart
        </i>
        <span
          className="cartNum"
        />
      </div>
    </div>
  </div>
  <div
    className="shipping-bar"
  >
    <h6>
      <i
        className="material-icons delivery"
        height="24"
        width="24"
      >
        local_shipping
      </i>
      <span
        className="underline"
      >
        FREE SHIPPING AND RETURNS
      </span>
      <img
        className="creators"
        height="24"
        src="https://s3-us-west-1.amazonaws.com/adidas-shoe/adidas-icon-creator.png"
        width="24"
      />
      <span
        className="underline"
      >
        CREATORS GET REWARDED - JOIN THE NEW ADIDAS CREATOR CLUB
      </span>
    </h6>
  </div>
</div>
`;
